inference-engine/src/transformations/src/transformations/common_optimizations/concat_reduce_fusion.cpp

```diff
@@ -0,0 +1,114 @@
+    auto reduce_axes_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto reduce_pattern = ngraph::pattern::wrap_type<opset8::ReduceMax>({concat_pattern, reduce_axes_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = std::dynamic_pointer_cast<opset8::Concat>(pattern_map.at(concat_pattern).get_node_shared_ptr());
+        int64_t concat_axis = concat->get_axis();
+
+        auto reduce = std::dynamic_pointer_cast<opset8::ReduceMax>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+        auto reduction_axes = reduce->get_reduction_axes();
+        auto reduction_axis = reduction_axes.to_vector()[0];
+        if (reduction_axes.size() != 1 || concat_axis != reduction_axis) {
+            return false;
+        }
+
+
+        auto concat_left_input = concat->get_input_node_shared_ptr(0);
+        auto concat_right_input = concat->get_input_node_shared_ptr(1);
+
+        const bool have_unsqueeze_on_inputs = std::dynamic_pointer_cast<ngraph::opset8::Unsqueeze>(concat_left_input) != nullptr &&
+                                              std::dynamic_pointer_cast<ngraph::opset8::Unsqueeze>(concat_right_input) != nullptr &&
+                                              concat_left_input->get_output_size() == 1 && concat_right_input->get_output_size() == 1;
+
+        const bool should_squeeze_finally = have_unsqueeze_on_inputs ? reduce->get_keep_dims() : !reduce->get_keep_dims();
+
+        if (have_unsqueeze_on_inputs) {
+            // TODO: Eliminate unsqueeze along both inputs
+            ngraph::replace_output_update_name(concat_left_input->output(0), concat_left_input->input_value(0));
+
+            ngraph::replace_output_update_name(concat_right_input->output(0), concat_right_input->input_value(0));
+        }
+        std::shared_ptr<ngraph::Node> result_node = register_new_node<ngraph::opset8::Maximum>(concat->get_input_node_shared_ptr(0),
+                                                                                              concat->get_input_node_shared_ptr(1));
+        if (should_squeeze_finally) {
+            auto squeeze_axis_node = ngraph::opset8::Constant::create(ngraph::element::i64, {}, {reduction_axis});
+            result_node = register_new_node<ngraph::opset8::Squeeze>(result_node, squeeze_axis_node);
+        }
+
+        result_node->set_friendly_name(reduce->get_friendly_name());
+
+        copy_runtime_info({concat,
+                           reduce,
+                          },
+                          result_node);
+        replace_node(reduce, result_node);
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(reduce_pattern, matcher_name);
+    register_matcher(m, callback);
+}
+
+ngraph::pass::ConcatReduceMinFusion::ConcatReduceMinFusion() {
```

Comment:
Use single transformation to handle all type of reductions. You can do it by specifying types in wrap_type:
```
 auto reduce_pattern = ngraph::pattern::wrap_type<opset8::ReduceMin, opset8::ReduceMax>({concat_pattern, reduce_axes_pattern});
```